spec/controllers/course/assessment/programming_evaluations_controller_spec.rb

```diff
@@ -52,5 +54,64 @@
         it { is_expected.to respond_with(400) }
       end
     end
+
+    describe '#load_and_authorize_pending_programming_evaluation' do
+      subject { controller.send(:load_and_authorize_pending_programming_evaluation) {} }
+
+      context 'when the allocated programming evaluation was assigned to another thread' do
+        before do
+          count = 0
+          wrapper = proc do |method, *args|
+            count += 1
+            raise IllegalStateError if count == 1
+            method.call(*args)
+          end
+
+          expect(controller).to receive(:find_pending_programming_evaluation).exactly(:twice).
```

Comment:
first try to force the failure, second try to succeed... otherwise the spec will never finish running :stuck_out_tongue: